InomialClient.js

```diff
@@ -33,13 +40,31 @@ class InomialClient {
     // Origin is nullable is the WSS Origin header; should be null for web clients, or
     // you can set it to let the server know what your application name is.
     //
+    // websocketClientConfig is an optional JS object passed-on verbatim to the WebSocketClient
+    // constructor; this can allow the caller to fine-tune socket/TLS parameters as needed.
+    // See <https://github.com/theturtle32/WebSocket-Node/blob/master/docs/WebSocketClient.md#client-config-options>
+    // for more details on what properties this object can accept.
     //
-    constructor(hostname, stage, origin, apikey)
+    // logLevel (if specified) should be one of the following constants:
+    //   LogLevel.NONE          No logging at all.
+    //   LogLevel.ERRORS        Only print errors (e.g. connection dropouts).
+    //   LogLevel.INFO          Also print informational messages (connection establishment).
+    //   LogLevel.FINE          Also print wire-level logging of GraphQL requests/responses.
+    // Default logging level is INFO if logLevel is null or omitted, but can be overridden with INOMIAL_LOG_LEVEL
+    // environment variable.
+    constructor(hostname, stage, origin, apikey, websocketClientConfig, logLevel)
     {
+        if (!hostname && !("INOMIAL_HOSTNAME" in process.env))
+          throw new Error("No hostname given (INOMIAL_HOSTNAME is unset)");
+        if (!stage && !("INOMIAL_STAGE" in process.env))
```

Comment:
if the `stage` can't be determined, default to `live`.

Reply:
To confirm: We still need to throw an error if no hostname given? (since it's not reasonable to improvise a default).